feature/add-task-bottom-sheet

abdo-essam opened the pull request.

AbdoMagdey24 committed:
feat: design categories screen with fixed data

Abdulrahman-Ragab-01 committed:
feat: add task bottom sheets details

Abdulrahman-Ragab-01 committed:
feat: add task bottom sheets details

abdo-essam committed:
feat: Add categories screen and UI models

This commit introduces the categories screen with its UI
models and required drawables.  It also adds the initial
implementation of the categories screen to allow users to
view categories and add new ones.

Abdulrahman-Ragab-01 committed:
feat: add AddOrEditTaskDetails on BottomSheet

Abdulrahman-Ragab-01 committed:
fix: pull priority functions

Abdulrahman-Ragab-01 committed:
feat: add priority section

abdo-essam committed:
feat: Implement Add/Edit Task Bottom Sheet

Replaced the CreateTaskScreen with a bottom sheet
implementation using TudeeBottomSheetScreen. Added
AddOrEditTaskDetails composable to encapsulate task
details input fields and PrioritySelector for task priority.
Also added new strings resources.

abdo-essam committed:
feat: add categories grid at task details bottom sheet ui

abdo-essam committed:
feat: refactor AddOrEditTaskDetails into smaller composable functions

abdo-essam committed:
feat: Add strings for task bottom sheet

Adds string resources for the task bottom sheet,
including labels for "Add", "Save", "Cancel", "Submit",
and "Select Date". Also includes strings for moving tasks
to in progress/done, and task details. These strings are
added to both the default and Arabic resource files.

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskBottomSheet.kt:
```diff
@@ -0,0 +1,84 @@
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+
```
Why this content is empty ?

marwanqashwa replied:
@AbdoMagdey24
i think it should be empy no colmun or any its only the buttom sheet and screen content will be empty to show the perant screen what do you think ?

Abdulrahman-Ragab-01 replied:
Yeah, it's just for testing the content will be the home screen when it is ready

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskBottomSheet.kt:
```diff
@@ -0,0 +1,84 @@
+            TudeeSecondaryButton(
+                modifier = Modifier.fillMaxWidth(),
+                text = stringResource(R.string.cancel),
+                onClick = { },
```
Why no action here ?

marwanqashwa replied:
and here

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskBottomSheet.kt:
```diff
@@ -0,0 +1,84 @@
+                modifier = Modifier.fillMaxWidth(),
+                text = stringResource(buttonText),
+                isDisabled = title.equals(null),
+                onClick = { },
```
Why no action here

marwanqashwa replied:
@AbdoMagdey24
i think it should be empy no colmun or any its only the buttom sheet and screen content will be empty to show the perant screen what do you think ?

Abdulrahman-Ragab-01 replied:
also it is only for testing until we make the RoomDB

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskDetails.kt:
```diff
@@ -0,0 +1,167 @@
+        icon = R.drawable.add_task_icon,
+        hint = R.string.task_title,
+        value = "",
+        onValueChange = {}
```
Why all these onValueChanged in this method with no action ? 
even if it doesn't have action now i think we should prepare it to be passed from outside the method

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskDetails.kt:
```diff
@@ -0,0 +1,167 @@
+}
+
+@Composable
+private fun CategoriesGrid(
```
Why we don't use the already existing screen of Categories ? I see it is the same design except the count of tasks

AbdoMagdey24 commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskDetails.kt:
```diff
@@ -0,0 +1,167 @@
+    selectedCategory: CategoryUiModel?,
+    onCategorySelected: (CategoryUiModel) -> Unit
+) {
+    SectionTitle(titleRes = R.string.category_title)
```
It should be Category not Category title as per Figma

abdo-essam replied:
Done

marwanqashwa commented on app/src/main/java/com/london/tudee/presentation/screens/tasks/AddOrEditTaskBottomSheet.kt:
```diff
@@ -0,0 +1,84 @@
+                modifier = Modifier.fillMaxWidth(),
+                text = stringResource(buttonText),
+                isDisabled = title.equals(null),
+                onClick = { },
```
i think you should upper the actions to make state hosting

marwanqashwa reviewed (commented):
please move all actions to top to make state hoisting

abdo-essam committed:
feat: Enhance Add/Edit Task Bottom Sheet with improved UI and functionality

abdo-essam committed:
feat: Add preview for CategorySection in AddOrEditTaskDetails

karimdeabes committed:
comp: add date picker component

abdo-essam committed:
feat: Implement task bottom sheet

Implements a bottom sheet for adding/editing tasks.
The sheet now dynamically adjusts its height based on
screen size and content, and uses a scrollable container
to prevent content overflow. Added bottom spacing to content.

abdo-essam committed:
feat: Refactor padding and update category title strings in AddOrEditTaskDetails

abdo-essam committed:
feat: Add ConfirmDeleteTaskScreen and SwipeToDeleteTask components with UI enhancements

abdo-essam committed:
feat: Add date picker functionality to AddOrEditTaskDetails and update TudeeTextField for read-only state

abdo-essam committed:
feat: Add preview for DatePicker component

This commit introduces a preview function for the `DatePicker` composable.
This allows for easier visualization and testing of the component within
Android Studio's preview pane.

Additionally, an unused import `androidx.compose.foundation.isSystemInDarkTheme`
was removed from `MainActivity.kt`.

abdo-essam committed:
feat: Update date format in AddOrEditTaskDetails and enhance TudeeTextField for read-only interaction

abdo-essam committed:
feat: Clean up comments and improve readability in AddOrEditTaskDetails

abdo-essam committed:
feat: Clean up comments and improve readability in AddOrEditTaskDetails

abdo-essam committed:
feat: Implement task bottom sheet and edit task screen

This commit introduces the task bottom sheet and edit task details
screen. It includes new drawables, modifies existing components like
DatePicker, Slider, StatusCard, and SwitchButton, and adds a new
screen for editing task details. Also includes translation string
changes. Removes redundant Arabic translation file.

Abdulrahman-Ragab-01 committed:
fix: add date picker from calender

abdo-essam committed:
fix: conflicts with development

abdo-essam committed:
refactor: Move Priority enum to domain layer

Moves the `Priority` enum from the presentation layer to
the domain layer. This improves code structure and promotes
better separation of concerns.
